Exception/JWTAuthenticationException.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Exception;
+
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+/**
+ * Exception to be used during a failed JWT authentication process.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class JWTAuthenticationException extends AuthenticationException
```

Comment:
I rely on the fact that the dispatched event should be enough to know the failure reason, but for instance we can't differentiate an invalid token from an expired one without checking `AuthenticationException::getMessage()` (see #223).
I'll spit that class in many 👍  each one extending `AuthenticationException` and overriding `getMessageKey()` (right?)